test/client.js

```diff
@@ -438,13 +456,44 @@ describe('CloudantClient', function() {
           done();
         });
       });
+
+      it('performs request with piped payload and returns response', function(done) {
+        var mocks = nock(SERVER)
+            .post(DBNAME + '/_all_docs', function(body) {
+              assert.deepEqual(body, { keys: [ 'doc1' ] });
+              return true;
+            })
+            .reply(200, { rows: [{ key: 'doc1', value: { rev: '1-xxxxxxxx' } }] });
+
+        var cloudantClient = new Client({ plugins: [] });
+        assert.equal(cloudantClient._plugins.length, 0);
+
+        var readable = new stream.Readable();
```

Comment:
you could also do:
```
var readable = new stream.PassThrough();
readable.end('{"keys":["doc1"]}');
```
I guess it saves a line, but it's not much of an optimisation.

Reply:
wil change.